In [1]:
%matplotlib notebook
import pandas as pd, numpy as np
import matplotlib.pyplot as plt
import sys

sys.path.append('../')
pd.options.display.max_columns = 35

In [2]:
data_path = '../data/raw/'

In [3]:
train = pd.read_csv(data_path + "train_30.csv", header=0, delimiter=";",decimal=',',
                    parse_dates=['date'], index_col='date')

In [4]:
train.head(5)

insee  tH2_obs  capeinsSOL0  ciwcH20  clwcH20 ddH10_rose4  \
date                                                                       
2014-01-01   6088001      7.8    38.245480      0.0      0.0         1.0   
2014-01-01  31069001      9.2    22.303022      0.0      0.0         3.0   
2014-01-01  33281001      9.3   194.861009      0.0      0.0         4.0   
2014-01-01  35281001      7.6   182.890703      0.0      0.0         3.0   
2014-01-01  59343001      9.9    31.526467      0.0      0.0         3.0   

               ffH10      flir1SOL0     fllat1SOL0     flsen1SOL0  flvis1SOL0  \
date                                                                            
2014-01-01  1.423250 -100292.923371 -108461.086720  -26163.859606         0.0   
2014-01-01  2.408082  -18815.426563  -11894.040428   14095.776354         0.0   
2014-01-01  4.535840  -61941.404781  -81870.097549  179862.881572         0.0   
2014-01-01  4.881681 -227613.396837    2019.225971  176067.945920         0.0   
2014-01-01  8.086883  -80526.229570  -76652.492578  192696.865313         0.0   

            hcoulimSOL0       huH2  iwcSOL0  nbSOL0_HMoy  nH20  ntSOL0_HMoy  \
date                                                                          
2014-01-01   503.778766  79.213137      0.0     5.857651   0.0     7.961908   
2014-01-01   729.086833  90.978967      0.0     8.000000   0.0     8.000000   
2014-01-01   897.634612  92.801612      0.0     7.744064   0.0     7.969099   
2014-01-01   262.459860  90.571364      0.0     3.776259   0.0     5.286151   
2014-01-01  1073.007502  88.712385      0.0     6.468433   0.0     8.000000   

                  pMER0   rr1SOL0     rrH20        tH2  tH2_VGrad_2.100  \
date                                                                      
2014-01-01  1014.035619 -0.012267  0.000000  11.657800        -0.856984   
2014-01-01  1010.331761  0.145260  0.008258   8.023305         1.454075   
2014-01-01  1006.380424  0.074460  0.018740  11.243986        -0.127558   
2014-01-01   999.551696  0.000000  0.000000   7.623182         0.243659   
2014-01-01   994.476250 -0.000781  0.000000   9.012843        -0.363116   

            tH2_XGrad  tH2_YGrad  tpwHPA850    ux1H10   vapcSOL0     vx1H10  \
date                                                                          
2014-01-01   0.004141  -0.003377   8.088612 -1.244766  20.734754  -4.553151   
2014-01-01   0.001222   0.000489  10.107755  1.112657  18.586576   3.962234   
2014-01-01   0.000368   0.000737  10.006774  7.714926  19.378744   8.029170   
2014-01-01   0.001054  -0.001233   7.359926  4.667206  13.485018   6.427848   
2014-01-01  -0.000285   0.000543   7.512743  6.852292  14.622403  11.882253   

            ech     mois  
date                      
2014-01-01   30  janvier  
2014-01-01   30  janvier  
2014-01-01   30  janvier  
2014-01-01   30  janvier  
2014-01-01   30  janvier

In [5]:
train.tail(1)

insee  tH2_obs  capeinsSOL0  ciwcH20  clwcH20 ddH10_rose4  \
date                                                                       
2016-05-30  75114001     12.5     2.818198      0.0      0.0         4.0   

               ffH10     flir1SOL0    fllat1SOL0    flsen1SOL0    flvis1SOL0  \
date                                                                           
2016-05-30  4.730778 -25714.018871 -91268.391242 -24893.136842  41743.419074   

            hcoulimSOL0       huH2  iwcSOL0  nbSOL0_HMoy  nH20  ntSOL0_HMoy  \
date                                                                          
2016-05-30   648.958456  87.014844      0.0          8.0   0.0          8.0   

                  pMER0   rr1SOL0     rrH20        tH2  tH2_VGrad_2.100  \
date                                                                      
2016-05-30  1008.250199  0.824805  0.080059  12.343202         -1.16857   

            tH2_XGrad  tH2_YGrad  tpwHPA850    ux1H10  vapcSOL0    vx1H10  \
date                                                                        
2016-05-30   0.000467    0.00131  14.466925  8.895326  30.78891 -3.814131   

            ech mois  
date                  
2016-05-30   30  mai

In [6]:
train.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 6167 entries, 2014-01-01 to 2016-05-30
Data columns (total 30 columns):
insee              6167 non-null int64
tH2_obs            6167 non-null float64
capeinsSOL0        5551 non-null float64
ciwcH20            6167 non-null float64
clwcH20            6041 non-null float64
ddH10_rose4        6041 non-null object
ffH10              6041 non-null float64
flir1SOL0          6160 non-null float64
fllat1SOL0         6160 non-null float64
flsen1SOL0         6160 non-null float64
flvis1SOL0         6160 non-null float64
hcoulimSOL0        6167 non-null float64
huH2               6041 non-null float64
iwcSOL0            6160 non-null float64
nbSOL0_HMoy        6160 non-null float64
nH20               5831 non-null float64
ntSOL0_HMoy        6160 non-null float64
pMER0              6034 non-null float64
rr1SOL0            6160 non-null float64
rrH20              6048 non-null float64
tH2                6041 non-null float64
tH2_VGrad_2.100  

ddH10_rose4, mois object type => category ?  
There are missing values

In [7]:
train.describe(include='all')

insee      tH2_obs  capeinsSOL0       ciwcH20       clwcH20  \
count   6.167000e+03  6167.000000  5551.000000  6.167000e+03  6.041000e+03   
unique           NaN          NaN          NaN           NaN           NaN   
top              NaN          NaN          NaN           NaN           NaN   
freq             NaN          NaN          NaN           NaN           NaN   
mean    4.390000e+07    10.296627    30.210074  1.116815e-07  1.295399e-05   
std     2.246953e+07     5.835699    96.864658  1.714985e-06  5.472359e-05   
min     6.088001e+06    -6.400000     0.000000  0.000000e+00  0.000000e+00   
25%     3.106900e+07     6.100000     0.036675  0.000000e+00  0.000000e+00   
50%     3.528100e+07    10.100000     1.534850  0.000000e+00  0.000000e+00   
75%     6.712400e+07    14.400000    11.841504  0.000000e+00  1.726422e-08   
max     7.511400e+07    29.400000  1237.521599  6.457207e-05  4.050511e-04   

       ddH10_rose4        ffH10      flir1SOL0    fllat1SOL0     flsen1SOL0  \
count         6041  6041.000000    6160.000000  6.160000e+03    6160.000000   
unique           4          NaN            NaN           NaN            NaN   
top            3.0          NaN            NaN           NaN            NaN   
freq          1875          NaN            NaN           NaN            NaN   
mean           NaN     2.987004 -164222.246876 -4.480046e+04   19691.762921   
std            NaN     1.946888  107340.093730  8.496074e+04   73137.022313   
min            NaN     0.028085 -522302.660251 -1.145786e+06 -229071.283900   
25%            NaN     1.617029 -249371.716476 -6.585640e+04  -19747.093793   
50%            NaN     2.534329 -182240.867188 -1.331940e+04    4873.812969   
75%            NaN     3.904950  -57922.039030  3.905953e+03   47356.991500   
max            NaN    14.908145   48134.305344  1.186144e+05  568788.919020   

           flvis1SOL0  hcoulimSOL0         huH2      iwcSOL0  nbSOL0_HMoy  \
count     6160.000000  6167.000000  6041.000000  6160.000000  6160.000000   
unique            NaN          NaN          NaN          NaN          NaN   
top               NaN          NaN          NaN          NaN          NaN   
freq              NaN          NaN          NaN          NaN          NaN   
mean    113917.870527   310.720918    84.230438     0.026781     3.289722   
std     174439.099716   346.850183    12.344721     0.373298     3.364038   
min          0.000000     1.838477    24.767455     0.000000     0.000000   
25%          0.000000    71.148288    78.679369     0.000000     0.002959   
50%       3483.362110   192.831584    87.586665     0.000000     1.824873   
75%     185317.924669   431.076599    92.966571     0.000000     7.343966   
max     687056.437069  5027.284656   100.028162    12.635832     8.000000   

               nH20  ntSOL0_HMoy        pMER0      rr1SOL0        rrH20  \
count   5831.000000  6160.000000  6034.000000  6160.000000  6048.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       0.542438     5.380670  1016.817542     0.073363     0.003224   
std        1.839701     3.232048     8.591797     0.383785     0.018154   
min        0.000000     0.000000   981.899875    -0.019531     0.000000   
25%        0.000000     1.875227  1011.889121     0.000000     0.000000   
50%        0.000000     7.524913  1017.054016     0.000000     0.000000   
75%        0.000000     8.000000  1022.235893     0.000836     0.000000   
max        8.230140     8.000000  1043.228797     9.247457     0.502367   

                tH2  tH2_VGrad_2.100    tH2_XGrad    tH2_YGrad    tpwHPA850  \
count   6041.000000      6041.000000  6041.000000  6041.000000  6160.000000   
unique          NaN              NaN          NaN          NaN          NaN   
top             NaN              NaN          NaN          NaN    

**tH2_obs** :	Observation de la température à 2 mètres in situ- au point station (prédictant)  
**capeinsSOL0**: 	Energie potentielle convective  
**ciwcH20**: 	Fraction de glace nuageuse à 20 mètres  
**clwcH20**: 	Fraction d'eau nuageuse à 20 mètres  
**ddH10_rose4**: 	Direction du vent à 10 mètres en rose4  
**ffH10**: 	Force du vent à 10 mètres en m/s  
**flir1SOL0**: 	Flux Infra-rouge en J/m2  
**fllat1SOL0**: 	Flux de chaleur latente en J/m2  
**flsen1SOL0**: 	Flux de chaleur sensible en J/m2  
**flvis1SOL0**: 	Flux visible en J/m2  
**hcoulimSOL0**: 	Hauteur de la couche limite en mètres  
**huH2**: 	Humidité 2mètres en %  
**iwcSOL0**: 	Réservoir neige kg/m2 (équivalent en eau liquide des chutes de neige)  
**nbSOL0_HMoy**: 	Nébulosité basse (moyenne sur les 6 points de grille autour de la station) (fraction en octat du ciel occulté)  
**nH20**: 	Fraction nuageuse à 20 mètres  
**ntSOL0_HMoy**: 	Nébulosité totale (moyenne sur les 6 points de grille autour de la station)  
**pMER0**: 	Pression au niveau de la mer  
**rr1SOL0**: 	Précipitation horaire au niveau du sol  
**rrH20**: 	Précipitation horaire à 20 mètres  
**tH2**: 	Température à 2 mètres du modèle AROME  
**tH2_VGrad_2.100**: 	Gradient vertical de température entre 2 mètres et 100 mètres  
**tH2_XGrad**: 	Gradient zonal de température à 2 mètres  
**tH2_YGrad**: 	Gradient méridien de température à 2 mètres  
**tpwHPA850**: 	Température potentielle au niveau 850 hPa  
**ux1H10**: 	Rafale 1 minute du vent à 10 mètres composante zonale  
**vapcSOL0**: 	Colonne de vapeur d'eau  
**vx1H10**: 	Rafale 1 minute du vent à 10 mètres composante verticale  
**ech** : 	Echéance de validité = date   

# Rows with nan

In [8]:
# Total missings values by variable
pd.DataFrame(train.isnull().sum()).T

insee  tH2_obs  capeinsSOL0  ciwcH20  clwcH20  ddH10_rose4  ffH10  \
0      0        0          616        0      126          126    126   

   flir1SOL0  fllat1SOL0  flsen1SOL0  flvis1SOL0  hcoulimSOL0  huH2  iwcSOL0  \
0          7           7           7           7            0   126        7   

   nbSOL0_HMoy  nH20  ntSOL0_HMoy  pMER0  rr1SOL0  rrH20  tH2  \
0            7   336            7    133        7    119  126   

   tH2_VGrad_2.100  tH2_XGrad  tH2_YGrad  tpwHPA850  ux1H10  vapcSOL0  vx1H10  \
0              126        126        126          7     126         0     126   

   ech  mois  
0    0     0

flir1SOL0 	fllat1SOL0 	flsen1SOL0 	flvis1SOL0 iwcSOL0 	nbSOL0_HMoy ntSOL0_HMoy rr1SOL0 tpwHPA850 : 7 missings  
rrH20 : 119 missings  
clwcH20 ddH10_rose4 ffH10 huH2 tH2 	tH2_VGrad_2.100 tH2_XGrad tH2_YGrad ux1H10 vx1H10 : 126 missings  
pMER0 : 133  
nH20 : 336  
capeinsSOL0 : 616 

In [9]:
for i in np.where(train.isnull().sum() == 7)[0]:
    print(np.where(np.asanyarray(pd.isnull(train.iloc[:,i]))))

(array([4627, 4628, 4629, 4630, 4631, 4632, 4633]),)
(array([4627, 4628, 4629, 4630, 4631, 4632, 4633]),)
(array([4627, 4628, 4629, 4630, 4631, 4632, 4633]),)
(array([4627, 4628, 4629, 4630, 4631, 4632, 4633]),)
(array([4627, 4628, 4629, 4630, 4631, 4632, 4633]),)
(array([4627, 4628, 4629, 4630, 4631, 4632, 4633]),)
(array([4627, 4628, 4629, 4630, 4631, 4632, 4633]),)
(array([4627, 4628, 4629, 4630, 4631, 4632, 4633]),)
(array([4627, 4628, 4629, 4630, 4631, 4632, 4633]),)


Same location for these missings vals

In [10]:
# Rows with missing val
np.where(train.isnull().sum(axis=1).values > 0)[0]

array([ 539,  540,  541,  542,  543,  544,  545,  546,  547,  548,  549,
        550,  551,  552,  553,  554,  555,  556,  557,  558,  559,  560,
        561,  562,  563,  564,  565,  566,  567,  568,  569,  570,  571,
        572,  573,  574,  575,  576,  577,  578,  579,  580,  581,  582,
        583,  584,  585,  586,  587,  588,  589,  590,  591,  592,  593,
        594,  595,  596,  597,  598,  599,  600,  601,  602,  603,  604,
        605,  606,  607,  608,  609,  610,  611,  612,  613,  614,  615,
        616,  617,  618,  619,  620,  621,  622,  623,  624,  625,  626,
        627,  628,  629,  812,  813,  814,  815,  816,  817,  818,  819,
        820,  821,  822,  823,  824,  825,  826,  827,  828,  829,  830,
        831,  832,  833,  834,  835,  836,  837,  838,  839, 1540, 1541,
       1542, 1543, 1544, 1545, 1546, 1547, 1548, 1549, 1550, 1551, 1552,
       1553, 1554, 1555, 1556, 1557, 1558, 1559, 1560, 1561, 1562, 1563,
       1564, 1565, 1566, 1567, 1568, 1569, 1570, 15

In [11]:
# Coordinate of missing values
np.where(np.asanyarray(pd.isnull(train)))

(array([ 539,  540,  541, ..., 4633, 4633, 4633]),
 array([ 2,  2,  2, ..., 24, 25, 27]))

Lets have a look to 1 row

In [12]:
pd.DataFrame(train.iloc[539+(7*11),:]).T

insee tH2_obs capeinsSOL0 ciwcH20 clwcH20 ddH10_rose4     ffH10  \
2014-03-30  6088001    11.3         NaN       0       0         1.0  0.559216   

           flir1SOL0 fllat1SOL0 flsen1SOL0 flvis1SOL0 hcoulimSOL0     huH2  \
2014-03-30   -319805   -39191.8    1483.23    83410.7      55.148  70.6249   

           iwcSOL0 nbSOL0_HMoy nH20 ntSOL0_HMoy    pMER0 rr1SOL0 rrH20  \
2014-03-30       0           0    0           8  1016.91       0     0   

                tH2 tH2_VGrad_2.100   tH2_XGrad   tH2_YGrad tpwHPA850  \
2014-03-30  13.4265        0.373662  0.00352482 -0.00676651   8.31097   

              ux1H10 vapcSOL0     vx1H10 ech  mois  
2014-03-30 -0.378848  11.1698 -0.0692918  30  mars

### Lets see if one can interpolate the missing values

In [13]:
for i in range(1,8):
    print(train.index[539-7*i], 
          ' - capeinsSOL0 val : {} for station {}'.format(train.capeinsSOL0[539-7*i], train.insee[539-7*i]))
print('interpolate : {}'.format(train.capeinsSOL0.interpolate()[539]))

2014-03-18 00:00:00  - capeinsSOL0 val : 7.71462916159 for station 6088001
2014-03-17 00:00:00  - capeinsSOL0 val : 7.90780714985 for station 6088001
2014-03-16 00:00:00  - capeinsSOL0 val : 0.0513649447693 for station 6088001
2014-03-15 00:00:00  - capeinsSOL0 val : 0.0677757209742 for station 6088001
2014-03-14 00:00:00  - capeinsSOL0 val : 0.0111768147617 for station 6088001
2014-03-13 00:00:00  - capeinsSOL0 val : 0.000466305369627 for station 6088001
2014-03-12 00:00:00  - capeinsSOL0 val : 0.00520225202489 for station 6088001
interpolate : 5.90483128579


In [14]:
for i in range(1,25):
    print(train.capeinsSOL0[539+7*i])

nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
0.206808648014
0.0500519458001
0.201991389069
0.0249546219275
0.0984533317242
0.0
5.07420018139
0.234053969383
2.89071020736
7.7248343433
69.6666567773
8.9741141126


### Difficile d'interpoler (ici pour capeinsSOL0)

In [15]:
for i in range(1,5):
    print(train.index[4627-7*i], 
          ' - flir1SOL0 val : {} for station {}'.format(train.flir1SOL0[4627-7*i], train.insee[4627-7*i]))
print('interpolate : {}'.format(train.flir1SOL0.interpolate()[4627]))
for i in range(1,5):
    print(train.flir1SOL0[4627+7*i])

2015-10-23 00:00:00  - flir1SOL0 val : -339351.45561 for station 6088001
2015-10-22 00:00:00  - flir1SOL0 val : -335334.200187 for station 6088001
2015-10-21 00:00:00  - flir1SOL0 val : -391646.537802 for station 6088001
2015-10-20 00:00:00  - flir1SOL0 val : -370097.530793 for station 6088001
interpolate : -165042.958353
-103793.381605
-100742.319785
-82790.7855024
-344353.275807


### Ici ca marche. A explorer

### Replace na val en utilisant le groupby. Faire des stats par city

https://machinelearningmastery.com/time-series-data-visualization-with-python/

In [16]:
# train.insee = train.insee.astype('category')
# groupby_city = train.groupby('insee')
# groupby_city.capeinsSOL0.apply(pd.Series.interpolate)

## Missing values by file

In [17]:
df_miss_val = pd.DataFrame(columns=train.columns)
for i in range(1,37):
    tmp =pd.read_csv(data_path + 'train_' + str(i) + '.csv', 
                     header=0, delimiter=";",decimal=',',
                     parse_dates=['date'], index_col='date')
    df_miss_val = pd.concat([df_miss_val, pd.DataFrame(tmp.isnull().sum()).T], axis=0 )
df_miss_val

insee tH2_obs capeinsSOL0 ciwcH20 clwcH20 ddH10_rose4 ffH10 flir1SOL0  \
0     0       0         581       7     119         126   126         7   
0     0       0         581       7     126         126   126        14   
0     0       0         581       7     119         126   126        14   
0     0       0         581       7     126         126   126        14   
0     0       0         581       7     119         126   126        14   
0     0       0         588       0     126         126   126        14   
0     0       0         588       7     119         126   126        14   
0     0       0         588       7     126         126   126        14   
0     0       0         588       0     126         126   126        14   
0     0       3         588       7     126         126   126        14   
0     0       2         595       0     119         126   126        14   
0     0       2         595       7     119         126   126        14   
0     0       2         602       0     119         126   126        14   
0     0       2         602       0     126         126   126        14   
0     0       1         602       7     119         126   126        14   
0     0       1         602       0     126         126   126        14   
0     0       2         602       0     119         126   126        14   
0     0       0         609       0     126         126   126        14   
0     0       2         609       7     126         126   126        14   
0     0       0         609       0     126         126   126        14   
0     0       0         609       0     126         126   126        14   
0     0       0         609       7     126         126   126        14   
0     0       0         609       7     119         126   126         7   
0     0       0         616       7     119         126   126         7   
0     0       0         616       7     126         126   126         7   
0     0       0         609       7     126         126   126         7   
0     0       0         616       7     126         126   126         7   
0     0       0         609       7     126         126   126         7   
0     0       0         609       7     126         126   126         7   
0     0       0         616       0     126         126   126         7   
0     0       0         133     140     126         133   133       140   
0     0       0         133     140     126         133   133       133   
0     0       0         133     140     133         133   133       133   
0     0       1         133     140     133         133   133       133   
0     0       1         133     140     133         133   133       133   
0     0       1         133     140     126         133   133       133   

  fllat1SOL0 flsen1SOL0 flvis1SOL0 hcoulimSOL0 huH2 iwcSOL0 nbSOL0_HMoy nH20  \
0          0          7          7           7  126       7           7  322   
0         14         14         14           7  126       7           7  329   
0          7         14         14           0  126       7           7  329   
0          7         14         14           7  126       7           7  329   
0          7         14         14           7  126       7           7  329   
0         10         14         14           7  126       7           7  329   
0          7         15         14           7  126       0           7  329   
0          7         14         14           7  119       7           7  329   
0          7         14         14           7  126       7           7  336   
0          7         15         14           7  119       7           0  336   
0          7         14         14           7  126       0           7  336   
0          7          7         14           7  126       7           7  336   
0          7          7         14           7  126       7           7  336   
0          7          7          7           7  126       7           7  336   
0          7          7   

# Link between temp and ech

## Data processing
Stack data

In [57]:
from os import listdir

temp_ech = None
cols = ['date', 'insee', 'tH2_obs', 'tH2', 'ech']
for file_ in listdir(data_path):
    if temp_ech is None:
        temp_ech = pd.read_csv(data_path + file_, header=0, 
                               delimiter=";",decimal=',',parse_dates=['date'], index_col='date', usecols=cols)
    else:
        temp_ech = pd.concat([temp_ech, 
                              pd.read_csv(data_path + file_, header=0, delimiter=";",decimal=',', 
                                          parse_dates=['date'], index_col='date', usecols=cols
                                         )])
# train.to_csv(data_path + "train.csv", sep=';', index=False)
temp_ech.sort_values(['insee', 'ech'], ascending=[True, True], inplace=True)
temp_ech.head(8)

insee  tH2_obs        tH2  ech
date                                        
2014-01-01  6088001      5.5   9.034338    1
2014-01-02  6088001      7.9  10.600003    1
2014-01-03  6088001      9.1  12.007224    1
2014-01-04  6088001     11.3  11.933810    1
2014-01-05  6088001     11.2  12.305508    1
2014-01-06  6088001      8.0  11.695416    1
2014-01-07  6088001      8.4  11.552058    1
2014-01-08  6088001     10.3  12.796752    1

## Analysis

In [59]:
print(max(temp_ech.index)-min(temp_ech.index))
temp_ech['2014-01-01']

880 days 00:00:00


insee  tH2_obs        tH2  ech
date                                         
2014-01-01   6088001      5.5   9.034338    1
2014-01-01   6088001      5.4   8.153351    2
2014-01-01   6088001      5.1   8.402343    3
2014-01-01   6088001      4.7   8.245682    4
2014-01-01   6088001      4.7   8.514988    5
2014-01-01   6088001      4.7   9.015853    6
2014-01-01   6088001      4.8   9.179396    7
2014-01-01   6088001      5.3   9.562431    8
2014-01-01   6088001      6.2  10.131776    9
2014-01-01   6088001      7.4  10.439922   10
2014-01-01   6088001     10.0  11.069077   11
2014-01-01   6088001     10.0  11.699109   12
2014-01-01   6088001     10.3  12.153840   13
2014-01-01   6088001     10.6  12.464955   14
2014-01-01   6088001      9.8  12.101505   15
2014-01-01   6088001      8.4  12.021394   16
2014-01-01   6088001      8.6  11.785885   17
2014-01-01   6088001      7.8  11.401471   18
2014-01-01   6088001      7.7  11.205267   19
2014-01-01   6088001      7.9  11.167496   20
2014-01-01   6088001      8.1  11.227059   21
2014-01-01   6088001      7.8  11.187069   22
2014-01-01   6088001      8.0  11.142321   23
2014-01-01   6088001      8.2  11.024646   24
2014-01-01   6088001      7.9  10.973166   25
2014-01-01   6088001      7.9  10.941004   26
2014-01-01   6088001      8.2  11.033950   27
2014-01-01   6088001      8.0  10.925451   28
2014-01-01   6088001      7.8  11.262741   29
2014-01-01   6088001      7.8  11.657800   30
...              ...      ...        ...  ...
2014-01-01  75114001      7.0   7.208069    1
2014-01-01  75114001      6.9   7.282375    2
2014-01-01  75114001      7.2   7.424495    3
2014-01-01  75114001      7.0   6.700273    4
2014-01-01  75114001      7.0   6.784263    5
2014-01-01  75114001      7.0   7.359250    6
2014-01-01  75114001      7.1   7.147274    7
2014-01-01  75114001      7.3   6.889038    8
2014-01-01  75114001      7.7   6.962505    9
2014-01-01  75114001      8.1   7.575566   10
2014-01-01  75114001      8.7   8.255433   11
2014-01-01  75114001      9.1   8.632080   12
2014-01-01  75114001      9.3   9.289964   13
2014-01-01  75114001      9.8   9.621031   14
2014-01-01  75114001     10.4   9.844250   15
2014-01-01  75114001     10.7  10.010461   16
2014-01-01  75114001     10.4   9.981056   17
2014-01-01  75114001      9.7   9.821576   18
2014-01-01  75114001      9.5   9.400470   19
2014-01-01  75114001      9.5   9.273802   20
2014-01-01  75114001     10.3   9.455015   21
2014-01-01  75114001     10.3   9.702389   22
2014-01-01  75114001     10.4  10.003702   23
2014-01-01  75114001     10.8   9.940145   24
2014-01-01  75114001     10.8  10.317167   25
2014-01-01  75114001     10.9   9.685958   26
2014-01-01  75114001      9.5  10.071367   27
2014-01-01  75114001      9.6  10.283924   28
2014-01-01  75114001     10.4   9.272826   29
2014-01-01  75114001     10.3   9.153009   30

[210 rows x 4 columns]

On peut voir que quand temp à n+1 with ech=1 == temp à n with ech=25 but not true with other cols

Dealing with missing val :  
https://gallery.cortanaintelligence.com/Experiment/Methods-for-handling-missing-values-1